builder-22/builder.toml

```diff
@@ -25,3 +33,13 @@ version = "0.14.0"
   [[order.group]]
     id = "heroku/nodejs"
     version = "0.5.4"
+
+[[order]]
+  [[order.group]]
+    id = "heroku/java-function"
```

Comment:
What detection order should we use for these?

With the PR as is, we'll have three different orders in use in our builders (even just considering the Java/Node.js buildpacks and ignoring the other buildpacks for a moment):

- `heroku/buildpacks:{18,20}`:
  -  ..., `heroku/nodejs-function`, `heroku/java-function`, `heroku/nodejs`, `heroku/java`
- `heroku/builder-classic:22`:
  -  ..., `heroku/java`, ..., `heroku/nodejs`
- `heroku/builder:22` (this PR):
  - `heroku/nodejs-function`, `heroku/nodejs`, `heroku/java-function`, `heroku/java`

Reply:
Yeah, it's inconsistent for sure. For the changes in this PR, I went with "copy what's already there" and replicated the order of the Node.js stuff. But since there was only one language in the builder, it's hard to derive intent from that. :)

I think the function buildpacks always need to come first since they would build just fine with the regular language buildpacks too - just without the language runtimes.

That leaves us with making decision how we order the languages themselves. Unless there is a technical reason for a specific order, alphabetical seems easy enough.

Reply:
@Malax The order can have a significant impact for projects that can match against multiple languages. The chance of this is higher for node.js where you'll get projects with both a front-end and backend component.

As to what order we should actually use, it's not clear :-)

Reply:
@edmorley That sounds like this is not a blocker for this PR? Where should we move this if at all?